From 473175ab3030092aa75c9e4e3daf81eb50da0278 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Tue, 4 Jan 2022 13:22:18 +0530
Subject: [PATCH] wpa_supplicant: support 5.9 channels in mesh 160mhz

currently, mesh supported 5.9 channels on 80Mhz. This patch supports
5.9 channels to bring up at 160mhz based on channel availability.

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 wpa_supplicant/wpa_supplicant.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: hostapd-2021-02-20-59e9794c/wpa_supplicant/wpa_supplicant.c
===================================================================
--- hostapd-2021-02-20-59e9794c.orig/wpa_supplicant/wpa_supplicant.c
+++ hostapd-2021-02-20-59e9794c/wpa_supplicant/wpa_supplicant.c
@@ -2511,7 +2511,7 @@ void ibss_mesh_setup_freq(struct wpa_sup
 	unsigned int bw80[] = { 5180, 5260, 5500, 5580, 5660, 5745, 5825, 5955,
 				6035, 6115, 6195, 6275, 6355, 6435, 6515,
 				6595, 6675, 6755, 6835, 6915, 6995 };
-	unsigned int bw160[] = { 5955, 6115, 6275, 6435, 6595, 6755, 6915 };
+	unsigned int bw160[] = { 5745, 5955, 6115, 6275, 6435, 6595, 6755, 6915 };
 	struct hostapd_channel_data *pri_chan = NULL, *sec_chan = NULL;
 	u8 channel;
 	int i, chan_idx, ht40 = -1, res, obss_scan = !(ssid->noscan);
@@ -2767,9 +2767,9 @@ skip_to_6ghz:
 	seg0 = freq->channel + 6;
 	seg1 = 0;
 
-/* setup center_freq1 for 6G 160MHz */
+	/* setup center_freq1 for 160MHz */
 	if ((mode->he_capab[ieee80211_mode].phy_cap[HE_PHYCAP_CHANNEL_WIDTH_SET_IDX] &
-	     HE_PHYCAP_CHANNEL_WIDTH_SET_160MHZ_IN_5G) && is_6ghz) {
+	     HE_PHYCAP_CHANNEL_WIDTH_SET_160MHZ_IN_5G)) {
 
 		for (j = 0; j < ARRAY_SIZE(bw160); j++) {
 			if (freq->freq == bw160[j]) {
@@ -2779,7 +2779,7 @@ skip_to_6ghz:
 				int channel = freq->channel + 16;
 
 				if (!ibss_mesh_is_80mhz_avail(channel, mode))
-					return;
+					break;
 
 				chwidth = CHANWIDTH_160MHZ;
 				seg0 = freq->channel + 14;
